Modify when memoryid is used

1814liyongbin opened the pull request:
<!--
Thank you so much for your contribution!

Please fill in all the sections below.
Please open the PR as a draft initially. Once it is reviewed and approved, we will ask you to add documentation and examples.
Please note that PRs with breaking changes or without tests will be rejected.

Please note that PRs will be reviewed based on the priority of the issues they address.
We ask for your patience. We are doing our best to review your PR as quickly as possible.
Please refrain from pinging and asking when it will be reviewed. Thank you for understanding!
-->

## Issue
<!-- Please specify the ID of the issue this PR is addressing. For example: "Closes #1234" or "Fixes #1234" -->
Closes #https://github.com/langchain4j/langchain4j/issues/984

## Change
<!-- Please describe the changes you made. -->
when the @MemoryId annotation is used, the chatMemoryProvider needs to be configured.

## General checklist
<!-- Please double-check the following points and mark them like this: [X] -->
- [ ✅] There are no breaking changes
- [✅ ] I have added unit and/or integration tests for my change
- [ ✅] The tests cover both positive and negative cases
- [✅ ] I have manually run all the unit and integration tests in the module I have added/changed, and they are all green
- [ ✅] I have manually run all the unit and integration tests in the [core](https://github.com/langchain4j/langchain4j/tree/main/langchain4j-core) and [main](https://github.com/langchain4j/langchain4j/tree/main/langchain4j) modules, and they are all green
<!-- Before adding documentation and example(s) (below), please wait until the PR is reviewed and approved. -->
- [ ] I have added/updated the [documentation](https://github.com/langchain4j/langchain4j/tree/main/docs/docs)
- [ ] I have added an example in the [examples repo](https://github.com/langchain4j/langchain4j-examples) (only for "big" features)
- [ ] I have added/updated [Spring Boot starter(s)](https://github.com/langchain4j/langchain4j-spring) (if applicable)


## Checklist for adding new maven module
<!-- Please double-check the following points and mark them like this: [X] -->
- [ ] I have added my new module in the root `pom.xml` and `langchain4j-bom/pom.xml`


## Checklist for adding new embedding store integration
<!-- Please double-check the following points and mark them like this: [X] -->
- [ ] I have added a `{NameOfIntegration}EmbeddingStoreIT` that extends from either `EmbeddingStoreIT` or `EmbeddingStoreWithFilteringIT`
- [ ] I have added a `{NameOfIntegration}EmbeddingStoreRemovalIT` that extends from `EmbeddingStoreWithRemovalIT`

## Checklist for changing existing embedding store integration
<!-- Please double-check the following points and mark them like this: [X] -->
- [ ] I have manually verified that the `{NameOfIntegration}EmbeddingStore` works correctly with the data persisted using the latest released version of LangChain4j

1814liyongbin committed:
feature: when the @MemoryId annotation is used, the chatMemoryProvider needs to be configured.

1814liyongbin committed:
feature: modification tips

1814liyongbin commented:
This is my first time contributing code to an open-source community, so please point out if there's anything inappropriate.

1814liyongbin committed:
feature: modification tips

1814liyongbin committed:
feature: modify code format

1814liyongbin committed:
feature: Optimize @MemoryId annotation and exception handling
- Pre-validate ChatMemoryProvider configuration
- Improve exception messages with configuration guidance
- Update unit tests
- Enhance MemoryId annotation documentation

Heezer commented on langchain4j/src/test/java/dev/langchain4j/service/AiServiceChatMemoryConfigTest.java:
```diff
@@ -0,0 +1,54 @@
+    }
+
+    @Test
+    void test_memoryId_requires_chatMemoryProvider_configuration_1() {
```
```suggestion
    void should_throw_exception_when_chat_memory_provider_is_not_configured() {
```

Heezer commented on langchain4j/src/test/java/dev/langchain4j/service/AiServiceChatMemoryConfigTest.java:
```diff
@@ -0,0 +1,54 @@
+    }
+
+    @Test
+    void test_memoryId_requires_chatMemoryProvider_configuration_2() {
```
```suggestion
    void should_return_response_when_chat_memory_provider_is_configured() {
```

Heezer commented on langchain4j/src/test/java/dev/langchain4j/service/AiServiceChatMemoryConfigTest.java:
```diff
@@ -0,0 +1,54 @@
+import org.mockito.junit.jupiter.MockitoExtension;
+
+@ExtendWith(MockitoExtension.class)
+public class AiServiceChatMemoryConfigTest {
```
```suggestion
class AiServiceChatMemoryConfigTest {
```

Heezer reviewed (changes requested):
good job, small suggestions provided

Heezer committed:
Merge branch 'main' into modify_when_memoryid_is_used

1814liyongbin committed:
Update langchain4j/src/test/java/dev/langchain4j/service/AiServiceChatMemoryConfigTest.java

ok

Co-authored-by: Alexey Titov <33568148+Heezer@users.noreply.github.com>

1814liyongbin committed:
Update langchain4j/src/test/java/dev/langchain4j/service/AiServiceChatMemoryConfigTest.java

Your advice has been very helpful to me, and I should be mindful of following naming conventions.

Co-authored-by: Alexey Titov <33568148+Heezer@users.noreply.github.com>

1814liyongbin committed:
Update langchain4j/src/test/java/dev/langchain4j/service/AiServiceChatMemoryConfigTest.java

I noticed that other test classes in the project are also not declared with public. Is it because test classes should not be accessed externally, and therefore they are not set to public? Thank you for your advice.

Co-authored-by: Alexey Titov <33568148+Heezer@users.noreply.github.com>

1814liyongbin committed:
refactor(langchain4j):mvn spotless:apply for MemoryId

1814liyongbin committed:
refactor(langchain4j):mvn spotless:apply for DefaultAiServices

1814liyongbin commented:
I've pulled this pull request from the main branch, and I only modified the necessary code. So why does other snippets of code indicate that they need to be formatted?

Heezer commented:
> I've pulled this pull request from the main branch, and I only modified the necessary code. So why does other snippets of code indicate that they need to be formatted?

I can imagine someone pushed to master unformatted code 😞
